samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -143,6 +146,9 @@ private Windows() { }
    */
   public static <M, K> Window<M, K, Collection<M>> keyedTumblingWindow(Function<M, K> keyFn, Duration interval) {
     BiFunction<M, Collection<M>, Collection<M>> aggregator = (m, c) -> {
+      if (c == null) {
+        return new ArrayList<>(Arrays.asList(m));
```

Comment:
1. +1
2. Don't we still need the branch anyways? (depending on whether `c` is null)
```
      if (c == null) {
        c = new ArrayList<>();
      }
      c.add(m);
      return c;

```
Do let me know if it can be re-written dropping the branch altogether.